src/utils.js

```diff
@@ -0,0 +1,85 @@
+import tmp from "tmp";
+import request from "request";
+import url from "url";
+import extract from "extract-zip";
+import fs from "fs";
+import { UI_STORE_FOLDER, ADMIN_PORT, HAPP_PORT } from "./constants";
+import { AdminWebsocket } from "@holochain/conductor-api";
+
+// Download from url to tmp file
+// return tmp file path
+const downloadFile = async (downloadUrl) => {
+    const fileName = tmp.tmpNameSync();
+    let file = fs.createWriteStream(fileName);
+
+    // Clean up url
+    let urlObj = new URL(downloadUrl);
+    urlObj.protocol = "https";
+    downloadUrl = urlObj.toString();
+
+    return new Promise((resolve, reject) => {
+        let stream = request({
+            uri: downloadUrl
+        })
+        .pipe(file)
+        .on('finish', () => {
+            console.log(`Downloaded file from ${downloadUrl} to ${fileName}`);
+            resolve(fileName);
+        })
+        .on('error', (error) => {
+            reject(error);
+        })
+    })
+}
+
+export const installDna = async (happ, agentPubKey) => {
+    const dnaPath = await downloadFile(happ.dna_url);
+    // Install via admin interface
+    let app, app_interface;
+    try {
+        const adminWebsocket = await AdminWebsocket.connect(
+            `ws://localhost:${ADMIN_PORT}`
+        );
+
+        app = await adminWebsocket.installApp({
```

Comment:
- The DNA doesn't get saved anywhere. Is this because it's handled automatically by Conductor-Api?
- Per https://github.com/Holo-Host/rfcs/pull/73/files we **eventually** need to set a UI password. I think it's not being checked right now so is it fine to leave blank?

Reply:
- Dna are saved to holochain's working db so they do need to persist in `/tmp` dir
- User authentication for self-hosted happs is omitted in this release. Next step will be implementing capabilities for self-hosted happs.